packages/logger/src/Logger.ts

```diff
@@ -253,6 +253,7 @@ class Logger implements ClassThatLogs {
 
     return (key, value) => {
       let item = value;
+      // noinspection SuspiciousTypeOfGuard
```

Comment:
`item` is just `value`, which is supposed to be `value: LogAttributes`? So maybe WebStorm linter is confused, but how can item then be an instance of `Error`?

<img width="736" alt="Screen Shot 2022-01-17 at 8 36 35 AM" src="https://user-images.githubusercontent.com/5442469/149807847-0e99cd46-fe38-4a72-9ae1-f43aeffa0c7b.png">

Reply:
Thanks. 

My understanding is that this function is called every time a log is emitted by `printLog` (see [L233](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/9c8506373cf328777eecab4dc09931ca0dbb6d87/packages/logger/src/Logger.ts#L233)) which means it might called with an error as argument like this:
```ts
try {
    throw new Error('hello');
  } catch (error) {
    logger.error('error:catch', error);  
  }
```

See debugger session from code above:
![image](https://user-images.githubusercontent.com/7353869/149819053-34b6a183-8545-4843-8eda-026898b95608.png)

Not sure why Webstorm is complaining (I use VS Code) but I would prefer addressing the issue (if there's one) rather than adding this comment.

Reply:
i will remove it for now. as maybe my understanding of typescript (or eslint) is still limited. So maybe it is a bug in Webstorm